packages/api/src/command/medical/patient/settings/quest-monitoring.ts

```diff
@@ -0,0 +1,170 @@
+  cxId,
+  patientIds,
+}: {
+  cxId: string;
+  patientIds: string[];
+}): Promise<PatientListProcessingResult> {
+  const { validPatientIds, invalidPatientIds: patientsNotFound } = await verifyPatients({
+    cxId,
+    patientIds,
+  });
+
+  async function batchProcessor(batch: string[]): Promise<void> {
+    await _addQuestSubscriptionToPatients({
+      patientIds: batch,
+      cxId,
+    });
+  }
+
+  const result = await processPatientsInBatches(validPatientIds, batchProcessor, {
+    cxId,
+    operationName: "addQuestSubscriptionToPatients",
+    errorMessage: "Failed to append Quest monitoring subscriptions for patients",
+    throwOnNoPatients: true,
+  });
+
+  return {
+    ...result,
+    patientsNotFound,
+  };
+}
+
+/**
+ * Removes a Quest monitoring subscription from the patient settings for the given customer and patient IDs.
+ *
+ * @param cxId The customer ID
+ * @param patientIds The patient IDs to remove Quest monitoring subscriptions for.
+ * @returns The number of patients updated and the list of patients not found
+ */
+export async function removeQuestSubscriptionFromPatients({
+  cxId,
+  patientIds,
+}: {
+  cxId: string;
+  patientIds: string[];
+}): Promise<PatientListProcessingResult> {
+  const { validPatientIds, invalidPatientIds: patientsNotFound } = await verifyPatients({
+    cxId,
+    patientIds,
+  });
+
+  async function batchProcessor(batch: string[]): Promise<void> {
+    await _removeQuestSubscriptionsFromPatients({
+      patientIds: batch,
+      cxId,
+    });
+  }
+
+  const result = await processPatientsInBatches(validPatientIds, batchProcessor, {
+    cxId,
+    operationName: "removeQuestSubscriptionFromPatients",
+    errorMessage: "Failed to remove Quest monitoring subscriptions for patients",
+    throwOnNoPatients: true,
+  });
+
+  return {
+    ...result,
+    patientsNotFound,
+  };
+}
+
+/**
+ * Adds a Quest monitoring subscription to patient settings if not already present
+ */
+async function _addQuestSubscriptionToPatients({
+  patientIds,
+  cxId,
+}: {
+  patientIds: string[];
+  cxId: string;
+}): Promise<void> {
+  const sequelize = PatientSettingsModel.sequelize;
+  if (!sequelize) {
+    throw new Error("Sequelize instance not available");
+  }
+
+  // Create patient settings records for patients who don't have them yet
+  await PatientSettingsModel.bulkCreate(
+    patientIds.map(patientId => ({
+      id: uuidv7(),
+      cxId,
+      patientId,
+      subscriptions: {
+        quest: true,
+      },
+    })),
+    {
+      ignoreDuplicates: true,
+    }
+  );
+
+  // Add Quest monitoring subscription to existing subscriptions (only if not already present)
+  // SQL explanation: If Quest monitoring already in array, keep current array, otherwise append Quest monitoring to array
+  const addSubscriptionQuery = `
+    UPDATE patient_settings 
+    SET 
+        subscriptions = jsonb_set(subscriptions, '{quest}', 'true'::jsonb, true),
```

Comment:
@lucasdellabella Is this the correct way to set another key on the `subscriptions` object without possibly overwriting an existing ADT subscription?